File: snippets/snippets/java/app/src/main/java/chatpost/Stream.java
/* (C)2024 */
package chatpost;

import com.cohere.api.Cohere;
import com.cohere.api.requests.ChatStreamRequest;
import com.cohere.api.types.ChatMessage;
import com.cohere.api.types.ChatTextGenerationEvent;
import com.cohere.api.types.Message;
import com.cohere.api.types.StreamedChatResponse;
import java.util.List;

public class Stream {
  public static void main(String[] args) {
    Cohere cohere = Cohere.builder().clientName("snippet").build();

    Iterable<StreamedChatResponse> response =
        cohere.chatStream(
            ChatStreamRequest.builder()
                .model("command-a-03-2025")
                .message("Tell me about LLMs")
                .build());

    for (StreamedChatResponse chatResponse : response) {
      if (chatResponse.isTextGeneration()) {
        System.out.println(
            chatResponse.getTextGeneration().map(ChatTextGenerationEvent::getText).orElse(""));
      }
    }

    System.out.println(response);
  }
}

File: snippets/snippets/node/chat-post/stream.js
import { CohereClient } from 'cohere-ai';

const cohere = new CohereClient({});

(async () => {
  const chatStream = await cohere.chatStream({
    model: 'command-a-03-2025',
    message: 'Tell me about LLMs',
  });

  for await (const message of chatStream) {
    if (message.eventType === 'text-generation') {
      process.stdout.write(message);
    }
  }
})();
